### PR TITLE
Deploy to avalanche, polygon and goerli

### PR DESCRIPTION
**Pull request by mikhailxyz**
part of https://github.com/gnosis/safe-react-apps/issues/229

The PR:
- adds deployments to goerli and avalanche
- new environment variable to opt-in for https://github.com/gnosis/safe-singleton-factory

### CODE REVIEW

File: allowances/networks.json [hunk 1]
```diff
@@ -1,1485 +1,1697 @@
 {
```
Comment (mikhailxyz) [inline allowances/networks.json:1]: truffle reformatted it. I am innocent

File: allowances/networks.json [hunk 2]
```diff
@@ -1,1485 +1,1697 @@
...
-            {
-              "indexed": false,
-              "internalType": "uint96",
-              "name": "payment",
-              "type": "uint96"
-            }
-          ],
-          "name": "PayAllowanceTransfer",
-          "type": "event"
+        "5": {
```
Comment (mikhailxyz) [inline allowances/networks.json:425]: new: goerli
https://goerli.etherscan.io/address/0xCFbFaC74C26F8647cBDb8c5caf80BB5b32E43134

File: allowances/networks.json [hunk 3]
```diff
@@ -1,1485 +1,1697 @@
...
-            {
-              "indexed": false,
-              "internalType": "address",
-              "name": "token",
-              "type": "address"
-            }
-          ],
-          "name": "DeleteAllowance",
-          "type": "event"
+        "43114": {
```
Comment (mikhailxyz) [inline allowances/networks.json:1272]: new: avalanche
https://snowtrace.io/address/0x1Fb403834C911eB98d56E74F5182b0d64C3b3b4D
https://snowtrace.io/tx/0x8f36f6b476c67938abdf3154a9c35c11c9379afd31d036ebf8faef12de1abe4d

File: allowances/scripts/generate_meta.js [hunk 4]
```diff
@@ -48,11 +48,11 @@ async function main() {
         const meta = JSON.parse(contractArtifact.metadata)
         etherscanConfig.language = meta.language
         etherscanConfig.evmVersion = meta.evmVersion
+        
         for (let source in meta.sources) {
             const pathParts = source.split("/")
-            const sourceFile = path.join(process.cwd(), source)
```
Comment (mikhailxyz) [inline allowances/scripts/generate_meta.js:53]: for me on macos the source already included cwd

File: allowances/networks.json [hunk 5]
```diff
@@ -1,1485 +1,1909 @@
...
+                            "name": "resetTime",
+                            "type": "uint16"
+                        }
+                    ],
+                    "name": "SetAllowance",
+                    "type": "event"
+                }
+            },
+            "links": {},
+            "address": "0x1Fb403834C911eB98d56E74F5182b0d64C3b3b4D",
```
Comment (mikhailxyz) [inline allowances/networks.json:1269]: new https://polygonscan.com/address/0x1fb403834c911eb98d56e74f5182b0d64c3b3b4d

**Review by yagopv — APPROVED** [review]

**Pull request merged**